src/arc_utilities/catches.py

```diff
@@ -0,0 +1,65 @@
+import signal
+from typing import Callable, Optional
+
+
+def catch_timeout(seconds: int, func: Callable, *args, **kwargs):
```

Comment:
Don't think we should have *args and **kwargs since it's not 100% clear how these are passed down (they get passed to the func, but they don't have to be). It's clearer to just take in a no-argument callable, which doesn't decrease the expressiveness since the user can just define a local function passing in those arguments. Thoughts on this?

Reply:
that's a fair point, although I find the method of declaring and extra local function to capture the args a little less elegant. I thought it was fairly obvious how args and kwargs would be used -- how about I rename them to `func_args` and `func_kwargs` to make it more clear?

Reply:
Sure